pydoctor/astbuilder.py

```diff
@@ -467,34 +476,60 @@ def visit_FunctionDef(self, node):
         self.default(node)
         self.builder.popFunction()
 
+    def _annotation_from_attrib(self, expr, ctx):
+        """Get the type of an C{attr.ib} definition.
+        @param expr: The expression's AST.
+        @param ctx: The context in which this expression is evaluated.
+        @return: A type annotation, or None if the expression is not
+                 an C{attr.ib} definition or contains no type information.
+        """
+        if isinstance(expr, ast.Call) \
+                and node2fullname(expr.func, ctx) in ('attr.ib', 'attr.attrib'):
+            keywords = {kw.arg: kw.value for kw in expr.keywords}
+            typ = keywords.get('type')
+            if typ is not None:
+                return self._unstring_annotation(typ)
+            default = keywords.get('default')
+            if default is not None:
+                return _infer_type(default)
+        return None
 
-def _annotation_from_attrib(expr, ctx):
-    """Get the type of an C{attr.ib} definition.
-    @param expr: The expression's AST.
-    @param ctx: The context in which this expression is evaluated.
-    @return: A type annotation, or None if the expression is not
-             an C{attr.ib} definition or contains no type information.
-    """
-    if isinstance(expr, ast.Call) \
-            and node2fullname(expr.func, ctx) in ('attr.ib', 'attr.attrib'):
-        keywords = {kw.arg: kw.value for kw in expr.keywords}
-        typ = keywords.get('type')
-        if typ is not None:
-            return _unstring_annotation(typ)
-        default = keywords.get('default')
-        if default is not None:
-            return _infer_type(default)
-    return None
+    def _unstring_annotation(self, node):
+        """Replace all strings in the given expression by parsed versions.
+        Returns the resulting node, or None if parsing failed.
+        """
+        try:
+            return _AnnotationStringParser().visit(node)
+        except SyntaxError as ex:
+            builder = self.builder
+            builder.system.msg(
+                'parsing',
+                '%s:%s: %s' % (
+                    builder.currentMod.description,
+                    node.lineno,
+                    'syntax error in annotation: %s' % ex),
```

Comment:
Why the nested `%` interpolation?

Reply:
It was copy-pasted code that I forgot to clean up once it worked. I'll replace it with something more readable.